src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -3436,5 +3436,413 @@ public void Deconstruct(out int a, out int b)
+                // (2,22): error CS1026: ) expected
+                // (string x, int y) = ("hello", 42);
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, @"""hello""").WithLocation(2, 22),
+                // error CS5001: Program does not contain a static 'Main' method suitable for an entry point
+                Diagnostic(ErrorCode.ERR_NoEntryPoint).WithLocation(1, 1)
+                );
+        }
+
+        [Fact]
+        public void NestedDeconstructionInScript()
+        {
+            var source =
+@"
+(string x, (int y, int z)) = (""hello"", (42, 43));
+System.Console.Write($""{x} {y} {z}"");
+";
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, expectedOutput: "hello 42 43");
+        }
+
+        [Fact]
+        public void VarDeconstructionInScript()
+        {
+            var source =
+@"
+(var x, var y) = (""hello"", 42);
+System.Console.Write($""{x} {y}"");
+";
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, expectedOutput: "hello 42");
+        }
+
+        [Fact]
+        public void NestedVarDeconstructionInScript()
+        {
+            var source =
+@"
+var (x1, (x2, x3)) = (""hello"", (42, 43));
+System.Console.Write($""{x1} {x2} {x3}"");
+";
+
+            Action<ModuleSymbol> validator = (ModuleSymbol module) =>
+            {
+                var sourceModule = (SourceModuleSymbol)module;
+                var compilation = sourceModule.DeclaringCompilation;
+                var tree = compilation.SyntaxTrees.First();
+                var model = compilation.GetSemanticModel(tree);
+
+                var x1 = GetDeconstructionVariable(tree, "x1");
+                var x1Symbol = model.GetDeclaredSymbol(x1);
+                Assert.Equal("System.String Script.x1", x1Symbol.ToTestDisplayString());
+                Assert.Equal("Field", x1Symbol.Kind.ToString());
+                var x1Field = (FieldSymbol)x1Symbol;
+                Assert.Equal("System.String", x1Field.Type.ToTestDisplayString());
+                Assert.Equal("Private", x1Field.DeclaredAccessibility.ToString());
+
+                var x2 = GetDeconstructionVariable(tree, "x2");
+                var x2Symbol = model.GetDeclaredSymbol(x2);
+                Assert.Equal("System.Int32 Script.x2", x2Symbol.ToTestDisplayString());
+                Assert.Equal("Field", x2Symbol.Kind.ToString());
+                var x2Field = (FieldSymbol)x2Symbol;
+                Assert.Equal("System.Int32", x2Field.Type.ToTestDisplayString());
+                Assert.Equal("Private", x2Field.DeclaredAccessibility.ToString());
+            };
```

Comment:
>}; [](start = 12, length = 2)

Consider verifying SemanticModel behavior for references to the fields (within the Write call).